.github/workflows/sync-main-to-preprod-and-prod.yml

```diff
@@ -83,7 +86,8 @@ jobs:
           check_filename() {
               if [[ $1 != "timestamp.json" && $1 != "snapshot.json" && !($1 =~ ^[0-9]+\.snapshot.json$) ]]; then
                   echo "Sync main to preprod and prod workflow: Files other than timestamp and snapshot were updated in main branch, including file: $1. Not syncing, exiting."
-                  exit 0
+                  echo "exit=2"  >> $GITHUB_OUTPUT
+                  exit 2
```

Comment:
Will this cause the workflow to show as failed?

Another way we could do this is split up the `sync` step into two: one which checks if any files besides the ts/snapshot have been updated, and if so, output something to `GITHUB_OUTPUT`, and a second step which executes only if that condition is not met. Then we wouldn't have to worry about error conditions.

Reply:
I like the idea of splitting up the sync step a lot better. I'll push that version soon.